glommio/src/executor/placement/mod.rs

```diff
@@ -178,14 +184,42 @@ impl Default for Placement {
 /// understand how `CpuSet` restrictions apply to each variant.  CPUs are
 /// identified via their [`CpuLocation`].
 #[derive(Clone, Debug, PartialEq)]
-pub struct CpuSet(Vec<CpuLocation>);
+pub struct CpuSet(HashSet<CpuLocation>);
+
+impl FromIterator<CpuLocation> for CpuSet
+{
+    fn from_iter<I: IntoIterator<Item = CpuLocation>>(cpus: I) -> Self {
+        let iter = cpus.into_iter();
+        let (_, size) = iter.size_hint();
+        let mut set = if let Some(size) = size {
+            HashSet::with_capacity(size)
+        } else {
+            HashSet::new()
+        };
+        for cpu in iter {
+            set.insert(cpu);
+        }
+        Self(set)
+    }
```

Comment:
Seems like it could be made a lot simpler by calling `from_iter()`.

not tested but something like this should work:

```suggestion
    fn from_iter<I: IntoIterator<Item = CpuLocation>>(cpus: I) -> Self {
        Self(HashSet<CpuLocation>::from_iter(cpu.into_iter()))
    }
```